src/server/services/procedures/geolocation/geolocation.js

```diff
@@ -50,23 +52,81 @@ async function reverseGeocode(lat, lon, query) {
     });
 }
 
-/**
- * Geolocates the address and returns the coordinates
- * @param {String} address target address
- * @returns {Object} structured data representing the location of the address
- */
-GeoLocationRPC.geolocate = async function (address) {
+async function rawGeolocate(address) {
     return await cache.wrap(address, async () => {
         logger.trace('Geocoding (not cached)', address);
         try {
             const res = await geocoder.geocode(address);
-            return { latitude: res[0].latitude, longitude: res[0].longitude };
+            if (res.length === 0 || !res[0]) throw Error('no results');
+            return res[0];
         } catch (e) {
             const message = `Geocoding (${address}) error: ${e.message}`;
             logger.warn(message);
             throw Error(`Failed to find location for ${address}`);
         }
     });
+}
+
+/**
+ * Geolocates the address and returns the coordinates
+ * @param {String} address target address
+ * @returns {Object} structured data representing the location of the address
+ */
+GeoLocationRPC.geolocate = async function (address) {
+    const raw = await rawGeolocate(address);
+    return { latitude: raw.latitude, longitude: raw.longitude };
+};
+
+/**
+ * Get information about the timezone of the provided address or location.
+ * You can provide either a list of two values representing the latitude and longitude location, or a string address to look up.
+ *
+ * @param {Union<Tuple<Latitude,Longitude>,String>} address target location - either a list representing [latitude, longitude], or an address string
+ * @returns {Object} information about the target's timezone
+ */
+GeoLocationRPC.timezone = async function (address) {
+    if (typeof(address) === 'string') {
+        address = await rawGeolocate(address);
+        address = [address.latitude, address.longitude];
+    }
+    const [latitude, longitude] = address;
+
+    return await cache.wrap(locString(latitude, longitude) + '.timezone', async () => {
+        logger.trace('timezone info (not cached)', latitude, longitude);
+        const res = await rp({
+            method: 'get',
+            uri: `${TIMEZONEDB_URL}/get-time-zone`,
+            qs: {
+                key: TimezoneDBKey.value,
```

Comment:
We should update this to allow users to provide their own API key (ie, add support for a service to use multiple keys). We should probably keep the convenience of the current approach - maybe add something like `setAuxiliaryAPIKey` or something. It could be accessed with `this.getAPIKey(type)` or something.

It seemed like a good idea to combine these but what would a timezone service look like? Are there enough RPCs to justify a standalone service?